runtime/router.go

```diff
@@ -78,9 +78,10 @@ type HandlerFn func(
 // RouterEndpoint struct represents an endpoint that can be registered
 // into the router itself.
 type RouterEndpoint struct {
-	EndpointName string
-	HandlerName  string
-	HandlerFn    HandlerFn
+	EndpointName  string
+	HandlerName   string
+	RawURLPattern string
```

Comment:
This means an endpoint is always associated with a url, what would be the answers to following questions:
- what is the expectation for below?
```
router.Handle("GET", "/foo",  http.HandlerFunc(zanzibar.NewRouterEndpoint(a, b, "foo", "foo", "/bar", c)))
```
it uses an endpoint that is for `/bar` to handle `/foo`, seems error-prone and shouldn't be allowed?
- previously I can use the same handler to handle different urls, now I will have to create one endpoint per url?

Overall, I don't think it is a good idea to couple the association between an url and an endpoint with the endpoint itself.

Reply:
I understand your points for concern.

However, I need a way to uniquely identify an endpoint in a form canonical to both edge gateway and rtapi. The two choices are the url or thrift method, since `EndpointName` and `HandlerName` are specific to edge gateway and not to rtapi. However, passing even the thrift method into `NewRouterEndpoint` would be coupling the association between a url and an endpoint with the endpoint itself because the thrift method is mapped 1-to-1 with the url.

I'll try to figure out a different way to uniquely identify an endpoint, but please let me know if you have any suggestions.

Again, thanks a lot for the review!